mcsema/BC/Function.cpp

```diff
@@ -260,6 +288,132 @@ static llvm::BasicBlock *GetOrCreateBlock(TranslationContext &ctx,
   return block;
 }
 
+static void LiftExceptionFrameLP(TranslationContext &ctx,
+                                 const NativeFunction *cfg_func) {
+
+  if (cfg_func->eh_frame.size() > 0) {
+
+    // The personality function `__gxx_personality_v0` is lifted as global variable. Erase the
+    // variable before declaring it as the function.
+    if (auto llvm_used = gModule->getGlobalVariable("__gxx_personality_v0")) {
+      llvm_used->eraseFromParent();
+    }
+
+    auto personality = gModule->getFunction("__gxx_personality_v0");
+
+    if (personality == nullptr) {
+      personality = llvm::Function::Create(
+          llvm::FunctionType::get(llvm::Type::getInt32Ty(*gContext), true),
+          llvm::Function::ExternalLinkage, "__gxx_personality_v0", gModule);
+    }
+
+    auto lifted_func = gModule->getFunction(cfg_func->lifted_name);
+    lifted_func->addFnAttr(llvm::Attribute::UWTable);
+    lifted_func->addFnAttr(llvm::Attribute::OptimizeNone);
+    lifted_func->removeFnAttr(llvm::Attribute::NoUnwind);
+#if LLVM_VERSION_NUMBER > LLVM_VERSION(3, 7)
+    lifted_func->setPersonalityFn(personality);
```

Comment:
is all of personality only relevant for LLVM 3.7? If yes, maybe guard the whole thing with this block, or do the other actions (like removing the `__gxx_personality` global variable still need to happen?

Reply:
The `llvm::Function` provide API to set the personality function after 3.7. In the previous version, the personality function was associated with the `landingpad`. I will refactor the code but removing the global variable is still needed.

Reply:
The checks here should be with LLVM 3.6.